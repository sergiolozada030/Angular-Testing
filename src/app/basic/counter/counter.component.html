<h2>Contador {{ counter }}</h2>
<hr />

<button type="button" class="btn btn-primary mx-2" (click)="increaseBy(1)">
  +1
</button>
<button type="button" class="btn btn-primary" (click)="increaseBy(-1)">
  -1
</button>
