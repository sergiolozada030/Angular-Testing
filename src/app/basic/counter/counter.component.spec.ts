import { ComponentFixture, TestBed } from '@angular/core/testing';
import { CounterComponent } from './counter.component';

describe('CounterComponent', () => {
  let component: CounterComponent;
  let fixture: ComponentFixture<CounterComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [CounterComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(CounterComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('Debe hacer match con el snapshot', () => {
    expect(compiled.innerHTML).toMatchSnapshot();
  });
});
