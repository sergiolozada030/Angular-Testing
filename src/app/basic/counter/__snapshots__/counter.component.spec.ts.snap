// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterComponent Debe hacer match con el snapshot 1`] = `
"<h2>Contador 0</h2><hr><button type=\\"button\\" class=\\"btn btn-primary mx-2\\"> +1
</button><button type=\\"button\\" class=\\"btn btn-primary\\"> -1
</button>"
`;
